### PR TITLE
TSK-1349: Added unit test for access items management

### PR DESCRIPTION
**Pull request by cnguyen-de**
https://taskana.atlassian.net/browse/TSK-1349

### CODE REVIEW

**Comment by sofie29** [comment]
There are some missing hooks in the auto-comment bot 🙂

File: web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    fixture = TestBed.createComponent(AccessItemsManagementComponent);
+    debugElement = fixture.debugElement;
+    app = fixture.debugElement.componentInstance;
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    app.accessId = { accessId: '', name: '' };
+    app.groups = [];
+  }));
+
+  it('should create the app', () => {
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts:66]: https://github.com/facebook/jest/blob/master/examples/angular/app.component.spec.ts 'it' is still used for tests in this example test

File: web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    const form = () => debugElement.nativeElement.querySelector('ng-form');
+    expect(form()).toBeFalsy();
+  });
+
+  it('app is initialized with ngxs store', () => {
+    store.selectOnce((state) => state).subscribe((state) => expect(state).toBeTruthy);
+    const engineConfigs = store.selectSnapshot((state) => state.engineConfiguration);
+    expect(engineConfigs).toBeTruthy();
+    const groups = store.selectSnapshot((state) => state.accessItemsManagement);
+    expect(groups).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts:90]: yes, before the component is init, there are no groups or engineConfigs attributes

File: web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    expect(typeAhead()).toBeTruthy();
+  });
+
+  it('result table should not display when search bar is empty', () => {
+    const form = () => debugElement.nativeElement.querySelector('ng-form');
+    expect(form()).toBeFalsy();
+  });
+
+  it('app is initialized with ngxs store', () => {
+    store.selectOnce((state) => state).subscribe((state) => expect(state).toBeTruthy);
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts:86]: when the code is promise-based, async should be used. When no promise is available, done() needs to be called, usually for subscription. So I'd use done() in this case.

File: web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    const engineConfigs = store.selectSnapshot((state) => state.engineConfiguration);
+    expect(engineConfigs).toBeTruthy();
+    const groups = store.selectSnapshot((state) => state.accessItemsManagement);
+    expect(groups).toBeTruthy();
+  });
+
+  it('onSelectedAccessId is called correctly', () => {
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts:93]: will be changed

File: web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+    app.onSelectAccessId(null);
+    expect(app.accessItemsForm).toBeNull();
+  });
+
+  it('GetAccessItems action is called in searchForAccessItemsWorkbaskets', async((done) => {
+    app.searchForAccessItemsWorkbaskets();
+    actions$.pipe(ofActionDispatched(GetAccessItems)).subscribe(async (action) => {
+      await expect(action).toBeTruthy();
+      done();
```
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts:109]: will look into it

File: web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts:66]: test and it behave the same, they are aliases: https://jestjs.io/docs/en/api.html#testname-fn-timeout
```
test(name, fn, timeout)
Also under the alias: it(name, fn, timeout)
```

File: web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (cnguyen-de) [reply web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts:109]: first async((done) is needed, then the 2nd async await isn't needed if done() is called already, code will be updated

File: web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (sofie29) [reply web/src/app/administration/components/access-items-management/access-items-management.component.spec.ts:125]: I thought you wanted to insert done(), didn't you?

**Pull request merged**